### DISCUSSION TITLE
Root user and regular user cannot communicate

### DISCUSSION THREAD

**Discussion by p-zzg** [eProsima/Fast-DDS#5261] [Q&A]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

Topic communication can cross users

### Current behavior

The subscriber starts using the root user, while the publisher starts using a regular user, but the subscriber cannot receive topic data

### Steps to reproduce

The subscriber starts using the root user, while the publisher starts using a regular user, but the subscriber cannot receive topic data

### Fast DDS version/commit

2.14.0

### Platform/Architecture

Ubuntu Focal 20.04 amd64

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

qos.transport().use_builtin_transports = false;
  auto shm_transport = std::make_shared < SharedMemTransportDescriptor > ();
  qos.transport().user_transports.push_back(shm_transport);
  auto udp_transport = std::make_shared< UDPv4TransportDescriptor >();
    udp_transport->non_blocking_send = true;
    udp_transport->interface_allowlist.emplace_back("lo", NetmaskFilterKind::OFF);
    udp_transport->interface_allowlist.emplace_back("eth2eth", NetmaskFilterKind::OFF);
    qos.transport().user_transports.push_back(udp_transport);

**Comment by JesusPoderoso** [comment]
Hi @Heyangq, thanks for using Fast DDS.
You are describing a known behavior in shared memory transport. Please refer to the [shared memory](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/shared_memory/shared_memory.html#segment) Fast DDS documentation section.
Launching any of the processes with a higher privileged user (for instance, root) can lead to communication problems, as processes run by non-privileged users may not be able to write into the memory segment.
Please, ensure that users run all the processes involved in the communication with the same privilege.
For that reason, I am moving this thread to the discussion forum as per the [Fast DDS contributing guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#issues-and-support).
